### DISCUSSION TITLE
C／C＋＋言語で簡単なプログラムを動かそう (連載2)no

### DISCUSSION THREAD

**Discussion by qmstake** [KyotoMicrocomputer/solid-rapi4#1] [Q&A] [answered]
「C／C＋＋言語で簡単なプログラムを動かそう (連載2)」を見ながら、Lチカができるか検証しようとしていますが、ビルドはできたものの、「Raspberry Pi 4Bボードにダウンロード」がどのような操作を行うことなのかが、よく分からず、困っております。ビルドしたファイルの実行方法（デバック実行ではなく実際の実行方法（コマンド入力等））を教えてください。
また、ビルドされた結果、実行ファイルが作成されると思いますが、その実行ファイルはどこに保存されており、Raspberry Pi 4Bで自動実行させるためにはどのような操作をしなくてはならないのか、教えてください。（最終的には、Raspberry Pi 4Bが起動したら、Raspberry Pi 4B上で実行ファイルが自動実行されるようにしなくてはならないので、そのやり方を教えてください。）
何分、RTOSを1から使うことが初めてなので、よろしくお願いいたします。

**Comment by KunihikoTsuji** [comment, marked as answer]
SOLID-IDEでビルドしたRTOS実行ファイルは、ソリューションファイル（拡張子 .ptsln）が存在するディレクトリ直下に、ビルド構成別に Debug/Release ディレクトリが生成され、そのディレクトリの中に拡張子 .slo というファイルで生成されています。

その slo ファイルを、scpコマンドや sftpコマンドを用いて、ラズパイの Linuxに転送してください。
例えば、以下のような感じです。

`scp cpp-blinky-rtos.slo pi@raspberrypi:
`

これで、Linuxファイルシステムに` /home/pi/cpp-blinky-rtos.slo` が追加されます。

RTOSプログラムのデバッガを使わない実行は、以下の URLに記載してあるので、これを参考にして試してみてください。
[https://github.com/KyotoMicrocomputer/solid-rapi4/blob/2ff10479b60d3b0ae0d748e6f11724b4e401e7d8/doc/system-linux.md#ghost-cli](https://github.com/KyotoMicrocomputer/solid-rapi4/blob/2ff10479b60d3b0ae0d748e6f11724b4e401e7d8/doc/system-linux.md#ghost-cli
)
